### ISSUE TITLE
Python - Spyder gets hang while using Pandas DataFrame

### ISSUE THREAD

**Issue by vrajs5** [spyder-ide/spyder#2713]
Recently I am facing serious issue with combination of spyder + pandas + ipython. 

I am using Spyder which is using iPython. I am trying following code which is working well:

```python
    import pandas as pd
    x = [list(range(5)) for i in range(1000)]
    pd.DataFrame(x)
```
But when I assigned same DataFrame to variable of same or different name it goes into hang phase (not coming out even after 30-45 minutes)

```python
    x = pd.DataFrame(x)
    ## OR
    y = pd.DataFrame(x)
```
I also found that assigning anything using pd.DataFrame or pd.concat to variable go in hang phase.

Same code is working well when I tried it in python console or ipython console.

Following are the related packages I have installed on my system. 

```python
    ipykernel                 4.0.3                    py27_0  
    ipython                   4.0.0                    py27_0  
    ipython-genutils          0.1.0                     <pip>
    ipython-notebook          4.0.4                    py27_0  
    ipython-qtconsole         4.0.1                    py27_0  
    ipython_genutils          0.1.0                    py27_0  
    ipywidgets                4.0.2                    py27_0  
    pandas                    0.16.1               np19py27_0  (Also tried 0.16.2)
    python                    2.7.10                        0  
    spyder                    2.3.6                    py27_0  
    spyder-app                2.3.6                    py27_0  
```

I am having Ubuntu 14.04 64 bit.

Let me know if you want more information. Same question is raised in SO http://stackoverflow.com/questions/32471150/python-spyder-gets-hang-while-using-pandas-dataframe

Thanks
Vishnu

**Comment by stonebig** [comment]
hi @vrajs5 ,

Is that what you tried ?

![spyder_3_0dev_pandas](https://cloud.githubusercontent.com/assets/4312421/10083631/931be3d8-62ff-11e5-9a21-df0c6d271abb.GIF)

**Comment by stonebig** [comment]
ah, on Legacy Python2.7 things are different (no variable shown):
![python27_spyder_2_3_6_pandas](https://cloud.githubusercontent.com/assets/4312421/10084216/8c9f7b84-6302-11e5-83b5-d6e40a7e0a40.GIF)

**Comment by sergeyf** [comment]
Just wanted to chime in that this is happening to me as well.

Running a script with this code makes Spyder hang (but it works in a separate Python console):

```
import pandas as pd

x = [[0,1,2],[3,4,5]]
df = pd.DataFrame(x)
```

**Comment by phsieh** [comment]
I've had the same issue running on Mac OS X 10.10.5
Entering 'pd.DataFrame(x)' into IPython returns the DataFrame without an issue;  
however,  assigning that same statement to a variable crashes spyder.

**Comment by stonebig** [comment]
today testing with Spyder 3.0dev on python 2.7 (windows):
==> I hang also .... worse than with 2.3.6

**Comment by stonebig** [comment]
no luck. maybe there won't be a working spyder on Python2.7 / IPython 4 / Windows environnement

**Comment by ccordoba12** [comment]
Yes, this problem only happens on Python 2. Closing as a duplicate of issue #2654.

**Comment by ccordoba12** [comment]
By the way, this will be fixed in 2.3.7.
